firmware/arduino/chessboard/movementFunctions.ino

```diff
@@ -1,72 +1,259 @@
+  subDeltaX = deltaX - (2 * diagDirX);
+  subDeltaY = deltaY - (2 * diagDirY);
+
+  // Calculate the direction of the diagonals
+  diagDirX = (deltaX > 0) ? 1 : -1;
+  diagDirY = (deltaY > 0) ? 1 : -1;
+
+  // If target point is equal to the start point
+  if (deltaX == 0  &&  deltaY == 0)
     return true;
+    
+  curCol = currPositionX / (stepsPerUnitSpace * 8);
+  curRow = currPositionY / (stepsPerUnitSpace * 8);
+
+  // Make initial move to first position. Move diagonally since it's faster.
+  if (!moveDirect(curCol, curRow, startCol, startRow))
+    return false;
+
+  // Calculate a list of up to 4 new points (including the start point) 
+  // We use previous points to calculate subsequent points, since they're on the same path
+  // Initial position
+  cols[0] = startCol;
+  rows[0] = startRow;
+
+  // Case where we should call moveDirect, since pathing can be simplified to that
+  if (absDeltaX <= 2  &&  absDeltaY <= 2)
+    return moveDirect(startCol, startRow, endCol, endRow);
+
+  // Because of the checks above, we know absDeltaX and absDeltaY can't both be 0 at the same time
+  // Case where we have a strictly vertical or horizontal movement along edges
+  if (absDeltaX == 0  ||  absDeltaY == 0)
+  {
+    uint8_t diagDirX2, diagDirY2;
+
+    if (absDeltaX == 0)
+    {
+      // Handle edge case where start position is on the edge of the board, so you must move inward
+      diagDirX = (startCol == (TOTAL_UNITSPACES - 1)) ? -1 : 1;
+
+      subDeltaX = 0;
+      subDeltaY = deltaY - (2 * diagDirY);  
+      diagDirX2 = -diagDirX;
+    }
+    else
+    {
+      // Handle edge case where start position is on the edge of the board, so you must move inward
+      diagDirY = (startRow == (TOTAL_UNITSPACES - 1)) ? -1 : 1;
+
+      subDeltaX = deltaX - (2 * diagDirX);
+      subDeltaY = 0; 
+      diagDirY2 = -diagDirY;
+    }
+
+    // Add diagonal movement
+    cols[1] = cols[0] + diagDirX;
+    rows[1] = rows[0] + diagDirY;
+
+    // Add straight movement
+    cols[2] = cols[1] + subDeltaX;
+    rows[2] = rows[1] + subDeltaY;
+
+    // Add diagonal movement, with flipped direction
+    cols[3] = cols[2] + diagDirX2;
+    rows[3] = rows[2] + diagDirY2;
+
+    // 3 added points, plus the initial point
+    numPoints = 4;
+  }
+  // Case where we're moving a cached piece to the graveyard from a capture
+  else if (startCol % 2 == 1  ||  startRow % 2 == 1)
+  {
+    subDeltaX = deltaX - diagDirX; 
+    subDeltaY = deltaY - diagDirY;
+
+    // Add straight X movement
+    cols[1] = cols[0] + subDeltaX;
+    rows[1] = rows[0];
+
+    // Add straight Y movement
+    cols[2] = cols[1];
+    rows[2] = rows[1] + subDeltaY;
+
+    // Add diagonal movement
+    cols[3] = cols[2] + diagDirX;
+    rows[3] = rows[2] + diagDirY;
+
+    // 3 added points, plus the initial point
+    numPoints = 4;
+  }
+  // Case where we have a knight or graveyard movement
+  else if ((absDeltaX == 4  &&  absDeltaY == 2)  ||  (absDeltaX == 2  &&  absDeltaY == 4))
```

Comment:
I gave you a bad conditional to check here. We want to check for cases where there is any amount of horizontal motion and `absDeltaY == 2` or any amount of vertical motion and `absDeltaX == 2`. So slightly more general than what is currently here. To that end, this should instead be
```c
if (absDeltaX == 2  ||  absDeltaY == 2)
```

If either is equal to 2, the other must be greater than equal to 2, because otherwise we would have hit one of the two cases we checked above
```c
if (absDeltaX <= 2  &&  absDeltaY <= 2)
if (absDeltaX == 0  ||  absDeltaY == 0)
```

Reply:
I agree with this, thanks for catching it!